File: app/src/main/java/com/github/overpass/gather/ui/meeting/chat/attachments/PhotosAdapter.java
package com.github.overpass.gather.ui.meeting.chat.attachments;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.github.overpass.gather.R;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class PhotosAdapter extends RecyclerView.Adapter<PhotosAdapter.AttachmentViewHolder> {

    private final List<String> photos = new ArrayList<>();
    private final OnClickListener listener;

    public PhotosAdapter(OnClickListener listener) {
        this.listener = listener;
    }

    @NonNull
    @Override
    public AttachmentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new AttachmentViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_photo, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull AttachmentViewHolder holder, int position) {
        holder.bind(photos.get(position), listener);
    }

    @Override
    public int getItemCount() {
        return photos.size();
    }

    public void setItems(@Nullable List<String> newPhotos) {
        if (newPhotos != null && !newPhotos.isEmpty()) {
            photos.clear();
            photos.addAll(newPhotos);
            notifyDataSetChanged();
        }
    }

    public class AttachmentViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.ivPhotoAttachment)
        ImageView ivPhotoAttachment;

        AttachmentViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

        public void bind(String photoUrl, OnClickListener listener) {
            Glide.with(itemView)
                    .load(photoUrl)
                    .into(ivPhotoAttachment);
            itemView.setOnClickListener((view) -> listener.onItemClick(photoUrl));
        }
    }

    public interface OnClickListener {
        void onItemClick(String photoUrl);
    }
}

File: app/src/main/java/by/overpass/gather/ui/meeting/chat/attachments/PhotosAdapter.kt
package by.overpass.gather.ui.meeting.chat.attachments

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import by.overpass.gather.R
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.item_photo.view.*
import java.util.*

class PhotosAdapter(
        private val listener: (String) -> Unit
) : RecyclerView.Adapter<PhotosAdapter.AttachmentViewHolder>() {

    private val photos = ArrayList<String>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AttachmentViewHolder {
        return LayoutInflater.from(parent.context)
                .run { inflate(R.layout.item_photo, parent, false) }
                .let { AttachmentViewHolder(it, listener) }
    }

    override fun onBindViewHolder(holder: AttachmentViewHolder, position: Int) {
        holder.bind(photos[position])
    }

    override fun getItemCount(): Int {
        return photos.size
    }

    fun setItems(newPhotos: List<String>?) {
        if (newPhotos != null && newPhotos.isNotEmpty()) {
            photos.clear()
            photos.addAll(newPhotos)
            notifyDataSetChanged()
        }
    }

    class AttachmentViewHolder(
            itemView: View,
            private val listener: (String) -> Unit
    ) : RecyclerView.ViewHolder(itemView) {

        private val ivPhotoAttachment: ImageView = itemView.ivPhotoAttachment

        fun bind(photoUrl: String) {
            Glide.with(itemView)
                    .load(photoUrl)
                    .into(ivPhotoAttachment)
            itemView.setOnClickListener { view -> listener(photoUrl) }
        }
    }
}
